AnkiDroid/src/main/java/com/ichi2/async/CollectionTask.java

```diff
@@ -1091,35 +1091,44 @@ protected StudyOptionsFragment.DeckStudyData task(@NonNull Collection col, @NonN
     }
 
     public static class ImportAdd extends TaskDelegate<String, Triple<AnkiPackageImporter, Boolean, String>> {
-        private final String mPath;
+        private final ArrayList<String> mPath;
 
 
-        public ImportAdd(String path) {
+        public ImportAdd(ArrayList<String> path) {
             this.mPath = path;
         }
 
 
         protected Triple<AnkiPackageImporter, Boolean, String> task(@NonNull Collection col, @NonNull ProgressSenderAndCancelListener<String> collectionTask) {
             Timber.d("doInBackgroundImportAdd");
             Resources res = AnkiDroidApp.getInstance().getBaseContext().getResources();
-            AnkiPackageImporter imp = new AnkiPackageImporter(col, mPath);
-            imp.setProgressCallback(new TaskManager.ProgressCallback(collectionTask, res));
-            try {
-                imp.run();
-            } catch (ImportExportException e) {
-                Timber.w(e);
-                return new Triple(null, true, e.getMessage());
-            }
+
+           AnkiPackageImporter imp = null; 
+
+            ArrayList<AnkiPackageImporter> impList;
+            // Iterate over mPath
+            for(int i=0; i<= mPath.size(); i++){
+
+                imp = new AnkiPackageImporter(col, mPath.get(i));
+                imp.setProgressCallback(new TaskManager.ProgressCallback(collectionTask, res));
+                try {
+                    imp.run();
+                } catch (ImportExportException e) {
+                    Timber.w(e);
+                    return new Triple(null, true, e.getMessage());
+                }
+            };
```

Comment:
https://github.com/ankidroid/Anki-Android/wiki/Code%20style
```suggestion
            for(int i = 0; i<= mPath.size(); i++) {

                imp = new AnkiPackageImporter(col, mPath.get(i));
                imp.setProgressCallback(new TaskManager.ProgressCallback(collectionTask, res));
                try {
                    imp.run();
                } catch (ImportExportException e) {
                    Timber.w(e);
                    return new Triple(null, true, e.getMessage());
                }
            };
```